kubeyard/commands/deploy.py

```diff
@@ -289,3 +298,55 @@ def upload_tarred_files(self, statics_tar_process):
             'socialwifi/aws-utils:1.0.0', 'upload_tar',
         ]
         self.docker_runner.run_with_output(statics_tar_process, *upload_statics_run_command)
+
+
+class AzureFilesStorage(FilesStorage):
+    azure_cli_image = 'microsoft/azure-cli:2.0.61'
+
+    def __init__(self, statics_directory, collect_statics_command, image, docker_runner, connection_string,
+                 bucket_name):
+        super().__init__(statics_directory, collect_statics_command, image, docker_runner)
+        self.connection_string = connection_string
+        self.bucket_name = bucket_name
+
+    def upload_tarred_files(self, statics_tar_process):
+        logger.info('Uploading to Azure...')
+        if sh.which('az'):
+            self._upload_with_local(statics_tar_process)
+        else:
+            self._upload_with_docker(statics_tar_process)
```

Comment:
After some team discussion please only leave the option: `_upload_with_docker`.

Reply:
Can you explain this a bit further?  What are the main reasons not to leave the user with the option to supply local tool?

From my perspective `az` is quite heavy, so in some environments (e.g. gitlab-ci) it really makes sense to make it part of that environment instead of spend significant time on fetching huge images.

Reply:
We just do not support local tools for this command.
When we use images, we are sure that it will work. When user have unsupported version of `az` installed locally, this will use it. We use only images to prevent this.

Reply:
After some discussion we decided to user `_upload_with_docker` by default. Option with `_upload_with_local` can be used with additional `--force-upload-with-local <path>` flag.